### PR TITLE
SYN-11 Added bigcommerce brand metafield and related methods.

### PR DESCRIPTION
**Pull request by abhisheksangani**

### CODE REVIEW

File: src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public String key;
+    public String value;
+    public String namespace;
+    public String permissionSet;
+    public String resourceType;
+    public Integer resourceId;
+    public String description;
+    public String dateCreated;
+    public String dateModified;
+
```
Comment (MlanLathiya) [inline src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java]: Remove use less line

File: src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafieldRequest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public String key;
+    public String value;
+    public String namespace;
+    public String permissionSet;
+    public String resourceType;
+    public Integer resourceId;
+    public String description;
+    public String dateCreated;
+    public String dateModified;
+
```
Comment (MlanLathiya) [inline src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafieldRequest.java]: Remove use less lines.

File: src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import lombok.Data;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class BrandMetafield {
+
+    public Integer id;
```
Comment (MlanLathiya) [inline src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java]: Set public to private in all.

**Review by MlanLathiya — CHANGES_REQUESTED** [review]

File: src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java (hunk 1 above)
Comment (abhisheksangani) [reply src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java]: ok

File: src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafieldRequest.java (hunk 2 above)
Comment (abhisheksangani) [reply src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafieldRequest.java]: ok

File: src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java (hunk 3 above)
Comment (abhisheksangani) [reply src/main/java/com/dft/bigcommerce/model/product/brandMetafields/BrandMetafield.java]: ok

**Review by manan-dft — APPROVED** [review]

**Review by MlanLathiya — APPROVED** [review]

**Pull request merged**